app/view/wechat.py

```diff
@@ -45,8 +50,17 @@ def wechat():
 
     if request.method == 'POST':
         xml_rec = et.fromstring(request.get_data())
-
         req = {k: xml_rec.find(k) for k in MSG_KEYS}
-        reply = get_reply(req['MsgType'], req['Content'])  # 根据信息类型获得回复
+
+        check_keys = ('MsgType', 'Content')
+        if not all(k in req for k in check_keys):
+            if req['MsgType'] == 'event':  # 接收事件推送
+                req['Content'] = xml_rec.find('Event')
+            else:
+                return error_jsonify(10000001)
+        elif not all(req[k] is not None for k in check_keys):
+            return error_jsonify(10000001)
```

Comment:
这句`if not all(k in req for k in check_keys):`应该是没有意义的，因为它检查`'MsgType', 'Content'`这两个键在`req`中的存在，然而`req`是由生成式`req = {k: xml_rec.find(k) for k in MSG_KEYS}`生成的，它必包含这两个键。这里应该做的是下面的，对`req`中的值进行判断的操作：
```suggestion
        if not all(req[k] is not None for k in check_keys):
            return error_jsonify(10000001)
        if req['MsgType'] == 'event':  # 接收事件推送
            req['Content'] = xml_rec.find('Event')
```
而且你以前写的逻辑相当于，如果MegType不是event，那就直接返回错误了

Reply:
Event类型消息的xml没有Content键，两个应该结合起来写，i get

Reply:
我大概猜到了，那即使这样的话，你也可以先判断MsgType的值，对Content赋值之后再使用上面的那个if判断。把我这两句if对调即可